libs/src/services/discoveryProvider/index.js

```diff
@@ -520,7 +523,10 @@ class DiscoveryProvider {
         // If disc prov is an unhealthy num blocks behind, retry with same disc prov with
         // hopes it will catch up
         console.info(`${this.discoveryProviderEndpoint} is too far behind. Retrying request at attempt #${attemptedRetries}...`)
```

Comment:
Nit: this log is a liar now :) Let's maybe log whether we retry or not in general?